### ISSUE TITLE
Add the support of the ZRANGESTORE command

### ISSUE THREAD

**Issue by infdahai** [apache/kvrocks#1460]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.

### Motivation

This issue considers to support the [zrangestore](https://redis.io/commands/zrangestore/) command.

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by jihuayu** [comment]
Hello, I can do this task. Can I submit a PR!

According to my understanding, I can use `RangeByRank` function to get elements from the src zset, and then use `Add` function to add them to the dst zset.

**Comment by git-hulk** [comment]
@jihuayu assigned,thanks!

**Comment by jihuayu** [comment]
In this command. I used `ZSet#RangeByLex` get members and use `ZSet#Add` add them to new zset.
But `ZSet#Add` need MemberScore(member+score), `ZSet#RangeByLex` only return member.

So I have 3 way to solve it.
1. After I use `ZSet#RangeByLex` get members, use `ZSet#Score` to get it score. It may cost additional performance overhead.
2. Add a new `ZSet#RangeByLex` function that return `MemberScore`. 
3. Change the `ZSet#RangeByLex` function make it return MemberScore. I found we have already get the socre in `ZSet#RangeByLex` function , so return `MemberScore` will not incur additional performance overhead.

I think 3rd is best. Is it OK?
